keras_cv/models/segmentation/segment_anything/sam.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from keras_cv.api_export import keras_cv_export
+from keras_cv.backend import keras
+from keras_cv.models.backbones.detectron2.detectron2_backbone import (
+    ViTDetBackbone,
+)
+from keras_cv.models.segmentation.segment_anything.sam_mask_decoder import (
+    SAMMaskDecoder,
+)
+from keras_cv.models.segmentation.segment_anything.sam_prompt_encoder import (
+    SAMPromptEncoder,
+)
+from keras_cv.models.task import Task
+
+
+@keras_cv_export("keras_cv.models.SegmentAnythingModel")
+class SegmentAnythingModel(Task):
```

Comment:
Let's add a preset to load up the fully-pretrained version of SAM.

Reply:
Done.